src/algorithm/native/to_wkt.rs

```diff
@@ -0,0 +1,282 @@
+use crate::geo_traits::*;
+
+/// Converts the input to WKT representation.
+pub trait ToWKT {
+    fn to_wkt(&self) -> String;
+}
+
+impl<G: GeometryTrait<T = f64>> ToWKT for G {
+    fn to_wkt(&self) -> String {
+        match self.as_type() {
+            GeometryType::Point(point) => {
+                let mut wkt = String::from("POINT");
```

Comment:
Instead of constructing `String` objects, would it be more performant to have a sort of "writer" concept? Every time we call `push` or `push_str` it allocates a new String right? I'm curious if emulating the Display trait https://doc.rust-lang.org/std/fmt/trait.Display.html might be useful?

Especially because we don't care that much about constructing _individual_ WKT buffers... we want to build up an array of string data, so if we could push all string data from all geometries into a single backing `String`/`Vec<u8>`, then that would be faster

Reply:
It is up to twice as fast as `wkt` and `geo`. Didn't consider how string arrays are generated.

Reply:
For fewer allocations, we could build a single `Vec<u8>` for the _entire_ string of the entire column, and keep track of the offsets for each row ourselves.

Reply:
If we could also reliably estimate the number of characters per generated WKT string, that would also make it more efficient for generating WKT from an entire array of geometries.

For now though, we can just use `StringBuilder::append_value`: https://docs.rs/arrow/latest/arrow/array/type.StringBuilder.html#method.append_value

Reply:
Went for the `Write` trait in `fmt`  in  https://github.com/geoarrow/geoarrow-rs/pull/788/commits/d3d586bf9a24a93b3dfab9e33518ab01dec9205c . It interfaces nicely with Arrows `StringBuilders` which can be constructed with capacity so we only allocate once in the best case we know how much.